From: "Alex Shi (Tencent)" <alexs@kernel.org>
Subject: mm/memcg: alignment memcg_data define condition
Date: Mon, 8 Jul 2024 14:32:36 +0800

commit 21c690a349ba ("mm: introduce slabobj_ext to support slab object
extensions") changed the folio/page->memcg_data define condition from
MEMCG to SLAB_OBJ_EXT.  The code works well, since config SLAB_OBJ_EXT is
fold into MEMCG in init/Kconfig.

But many related functions that deal with memcg_data still defined under
MEMCG instead of SLAB_OBJ_EXT, and FOLIO_MATCH with memcg_data are defined
under MEMCG too.  That looks weird and incorrect with memcg_data raw
meaning.

So let's put memcg_data under MEMCG config to alignment the definition
with FOLIO_MATCH and its usage in functions.

Link: https://lkml.kernel.org/r/20240708063236.1096395-1-alexs@kernel.org
Signed-off-by: Alex Shi (Tencent) <alexs@kernel.org>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Kees Cook <keescook@chromium.org>
Cc: Pasha Tatashin <pasha.tatashin@soleen.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm_types.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/include/linux/mm_types.h~mm-memcg-alignment-memcg_data-define-condition
+++ a/include/linux/mm_types.h
@@ -180,7 +180,7 @@ struct page {
 	/* Usage count. *DO NOT USE DIRECTLY*. See page_ref.h */
 	atomic_t _refcount;
 
-#ifdef CONFIG_SLAB_OBJ_EXT
+#ifdef CONFIG_MEMCG
 	unsigned long memcg_data;
 #endif
 
@@ -343,7 +343,7 @@ struct folio {
 			};
 			atomic_t _mapcount;
 			atomic_t _refcount;
-#ifdef CONFIG_SLAB_OBJ_EXT
+#ifdef CONFIG_MEMCG
 			unsigned long memcg_data;
 #endif
 #if defined(WANT_PAGE_VIRTUAL)
_
